core/src/edu/cornell/gdiac/main/controller/AudioController.java

```diff
@@ -0,0 +1,252 @@
+    }
+  }
+
+  public void initBGM(AudioSource source) {
+    AudioEngine engine = (AudioEngine) Gdx.audio;
+    bgMusic = engine.newMusicBuffer(false, SAMPLE_RATE);
+    bgMusic.addSource(source);
+    bgMusic.setLoopBehavior(true);
+  }
+
+  public void initBGM() {
+    AudioEngine engine = (AudioEngine) Gdx.audio;
+    bgMusic = engine.newMusicBuffer(false, SAMPLE_RATE);
+    bgMusic.setLoopBehavior(true);
+  }
+
+  /**
+   * Support for multiple soundtracks in the future
+   *
+   * @param source: the new audio source to be added to music queue
+   */
+  public void addBGM(AudioSource source) {
+    bgMusic.addSource(source);
+  }
+
+  /**
+   * Play sound effect based on types
+   *
+   * @param type: type of sound effect that needs to be played
+   */
+  public void playSFX(SFXtype type) {
+    // make sure sound is only played when game is not muted
+    if (!isSoundOn) {
+      return;
+    }
+    switch (type) {
+      case JUMP:
+        if (jumpSound != null) {
+          jumpId = playSound(jumpSound, jumpId);
+        }
+        break;
+      case BOUNCE:
+        if (bounceSound != null) {
+          bounceId = playSound(bounceSound, bounceId);
+        }
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported sound effect type: " + type);
+    }
+  }
+
+  public void playBGM() {
+    bgMusic.play();
+  }
+
+  /**
+   * Play all collision sound effects based on current contacts
+   *
+   * @param contacts: all contacts within the level, passed down by GameController
+   */
+  public void playCollision(Array<Contact> contacts) {
+    if (!isSoundOn) {
+      return;
+    }
+
+    boolean isSFXContact = false;       // Check if there is any contact that requires sfx
+    for (Contact contact : contacts) {
+      // NOTE: this might need refactoring into CollisionController if game performance drops
+      if (contact.isTouching()) {
+        Fixture fix1 = contact.getFixtureA();
+        Fixture fix2 = contact.getFixtureB();
+        Body body1 = fix1.getBody();
+        Body body2 = fix2.getBody();
+        Obstacle bd1 = (Obstacle) body1.getUserData();
+        Obstacle bd2 = (Obstacle) body2.getUserData();
+        // Check if any of the contact requires sfx
+        if (bd2 instanceof PlayerModel && bd1 instanceof BouncePlatformModel) {
+          isSFXContact = true;
+          if (!isSFXplayed) {
+            playSFX(SFXtype.BOUNCE);
+            isSFXplayed = true;
+          }
+        }
+      }
+    }
+    // Reset isSFXplayed if we are not currently touching any objects that requires sfx
+    if (!isSFXContact) {
+      isSFXplayed = false;
+    }
+  }
+
+  /**
+   * Temporarily pause all sounds for pausing menu
+   */
+  public void pause() {
+    if (jumpSound != null) {
+      if (jumpSound.isPlaying(jumpId)) {
+        jumpSound.stop(jumpId);
+      }
+    }
+    if (bounceSound != null) {
+      if (bounceSound.isPlaying(bounceId)) {
+        bounceSound.stop(bounceId);
+      }
+    }
```

Comment:
Could you do this with a loop instead? This way if we add more sounds we won't have to keep updating this method.